src/shared/mkfs-util.c

```diff
@@ -121,6 +123,14 @@ static int do_mcopy(const char *node, const char *root) {
         if (r < 0)
                 return log_error_errno(r, "Failed to determine whether mcopy binary exists: %m");
 
+        /* Avoid failures caused by mismatch in expectations between mkfs.vfat and mcopy by disabling
+        * the stricter mcopy checks using MTOOLS_SKIP_CHECK. */
+        source_date_epoch = strv_find_prefix(environ, "SOURCE_DATE_EPOCH=");
```

Comment:
hmm? why go via environ[]? why not just getenv(). does pretty much the same

Reply:
We want to have the full `SOURCE_DATE_EPOCH=1234`

Reply:
oh, right, of course